tests/test_OutputStream.cpp

```diff
@@ -1730,3 +1732,713 @@ BOOST_AUTO_TEST_CASE(Formatted_Existing)
 }
 
 BOOST_AUTO_TEST_SUITE_END() // Class_RFT
+
+// ==========================================================================
+
+BOOST_AUTO_TEST_SUITE(Class_SummarySpecification)
+
+namespace {
+    std::time_t advance(const std::time_t tp, const double sec)
+    {
+        using namespace std::chrono;
+
+        using TP      = time_point<system_clock>;
+        using DoubSec = duration<double, seconds::period>;
+
+        const auto t = system_clock::from_time_t(tp) +
+            duration_cast<TP::duration>(DoubSec(sec));
+
+        return system_clock::to_time_t(t);
+    }
+
+    std::time_t makeUTCTime(const std::tm& timePoint)
+    {
+        auto       tp    =  timePoint; // Mutable copy.
+        const auto ltime =  std::mktime(&tp);
+        auto       tmval = *std::gmtime(&ltime); // Mutable.
+
+        // offset =  ltime - tmval
+        //        == #seconds by which 'ltime' is AHEAD of tmval.
+        const auto offset =
+            std::difftime(ltime, std::mktime(&tmval));
+
+        // Advance 'ltime' by 'offset' so that std::gmtime(return value) will
+        // have the same broken-down elements as 'tp'.
+        return advance(ltime, offset);
+    }
+
+    std::string noWGName()
+    {
+        return ":+:+:+:+";
+    }
+
+    int noNum() { return 0; }
+
+    Opm::EclIO::OutputStream::SummarySpecification::StartTime
+    start(const int year, const int month, const int day,
+          const int hour, const int minute, const int second)
+    {
+        using std::chrono::system_clock;
+
+        auto timepoint = std::tm {};
+
+        timepoint.tm_sec  = second;
+        timepoint.tm_min  = minute;
+        timepoint.tm_hour = hour;
+        timepoint.tm_mday = day;
+        timepoint.tm_mon  = month - 1;
+        timepoint.tm_year = year - 1900;
+
+        return system_clock::from_time_t(makeUTCTime(timepoint));
+    }
+
+    Opm::EclIO::OutputStream::SummarySpecification::RestartSpecification
+    noRestart()
```

Comment:
I don't think the `RESTART`keyword is required in the SMSPEC file, so rather than writing an invalid `{"", -1}` I would recommend not writing it at all.

Reply:
Okay.  I've pushed an update to ensure we don't write `RESTART` to the SMSPEC file unless there actually is useful data to output.